src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -661,6 +662,24 @@ private bool Get_CORINFO_METHOD_INFO(MethodDesc method, MethodIL methodIL, CORIN
             Get_CORINFO_SIG_INFO(method, sig: &methodInfo->args, methodIL);
             Get_CORINFO_SIG_INFO(methodIL.GetLocals(), &methodInfo->locals);
 
+#if READYTORUN
+            if ((methodInfo->options & CorInfoOptions.CORINFO_GENERICS_CTXT_MASK) != 0)
+            {
+                foreach (var region in exceptionRegions)
+                {
+                    if (region.Kind == ILExceptionRegionKind.Catch)
+                    {
+                        TypeDesc catchType = (TypeDesc)methodIL.GetObject(region.ClassToken);
+                        if (catchType.IsRuntimeDeterminedSubtype)
```

Comment:
The IL is not RuntimeDetermined (most of the time) in JitInterface because RyuJIT can't handle it. I think this will fix the test failure:

```suggestion
                        if (catchType.IsCanonicalSubtype(CanonicalFormKind.Any))
```